arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1240,6 +1731,90 @@ void RocksDBCollection::adjustNumberDocuments(transaction::Methods& trx, int64_t
   meta().adjustNumberDocuments(seq, /*revId*/ 0, diff);
 }
 
+Result RocksDBCollection::upgrade() {
+  Result res{};
+  if (_logicalCollection.version() >= LogicalCollection::Version::v37) {
+    return res;
+  }
+
+  auto& server = _logicalCollection.vocbase().server();
+  auto& selector = server.getFeature<EngineSelectorFeature>();
+  RocksDBEngine& engine = selector.engine<RocksDBEngine>();
+
+  auto cleanupGuard = arangodb::scopeGuard(
+      [this]() -> void { [[maybe_unused]] Result res = cleanupAfterUpgrade(); });
+
+  res = ::updateObjectIdsForCollection(*engine.db(), _logicalCollection,
+                                       ::injectNewTemporaryObjectId, false);
+  if (res.fail()) {
+    return res;
+  }
+
+  {
+    // start an exclusive transaction to block access to the collection
+    std::shared_ptr<transaction::Context> context =
+        transaction::StandaloneContext::Create(_logicalCollection.vocbase());
+    SingleCollectionTransaction trx(context, _logicalCollection, AccessMode::Type::EXCLUSIVE);
+    res = trx.begin();
+    if (res.fail()) {
+      return res;
+    }
+    auto trxGuard = scopeGuard([&trx]() -> void { trx.abort(); });
```

Comment:
I don't think we need this guard. Transactions are automatically aborted when they go out of scope and there hasn't been an explicit commit.